pkg/transforms/http_test.go

```diff
@@ -91,6 +90,71 @@ func TestHTTPPost(t *testing.T) {
 	}
 }
 
+func TestHTTPPostWithSecrets(t *testing.T) {
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+
+		if r.URL.EscapedPath() == badPath {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+
+		readMsg, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if strings.Compare((string)(readMsg), msgStr) != 0 {
+			t.Errorf("Invalid msg received %v, expected %v", readMsg, msgStr)
+		}
+		//this should be here on successful secret retrieval
+		if r.Header.Get("Secret-Header-Name") != "value" {
+			t.Errorf("Unexpected header name received %s, expected %s", r.Header.Get("Content-type"), "application/json")
+		}
+		if r.Header.Get("Content-type") != "application/json" {
```

Comment:
use assert.Equal

Reply:
I don't really want to do that - in the case that no secrets are used then the header wont exist and i dont want it to fail since we have a test for ensuring that no call to the vault happens when no path and key are provided

Reply:
I am just saying use assert.Equal(..) instead if .. != ... It simplifies the code...